### PR TITLE
[generator] Use new ObsoletedOSPlatformAttribute for generated code

### PR DESCRIPTION
**Pull request by chamons**
- Part of https://github.com/xamarin/xamarin-macios/issues/15849
- A bit of dead code was removed while I was reviewing things.

### CODE REVIEW

**Comment by chamons** [comment]
I reviewed examples of the generated code and found no changes in legacy and what looked reasonable in NET builds.

File: src/generator-attributes.cs [hunk 1]
```diff
@@ -971,6 +971,13 @@ void GenerateUnsupported (StringBuilder builder)
 		builder.AppendLine ("\")]");
 	}
 
+	void GenerateDeprecated (StringBuilder builder)
+	{
+		builder.Append ("[ObsoletedOSPlatform (\"");
+		GeneratePlatformNameAndVersion (builder);
```
Comment (rolfbjarne) [inline src/generator-attributes.cs:977]: Looking at the generator diff, the `Message` part of the attribute is gone:

```diff
diff --git a/old/dotnet/ios/generated-sources/ARKit/ARFrame.g.cs b/new/dotnet/ios/generated-sources/ARKit/ARFrame.g.cs
index 21c8f0e..b80f9c6 100644
--- a/old/dotnet/ios/generated-sources/ARKit/ARFrame.g.cs
+++ b/new/dotnet/ios/generated-sources/ARKit/ARFrame.g.cs
@@ -151,16 +151,10 @@ namespace ARKit {
 		#endif
 		}
 		[Export ("hitTest:types:")]
-		#if __IOS__
-		[Advice ("Starting with ios14.0 Use 'ARSession.Raycast' instead.")]
-		#endif
-		[UnsupportedOSPlatform ("ios14.0")]
+		[ObsoletedOSPlatform ("ios14.0")]
 		[UnsupportedOSPlatform ("tvos")]
 		[UnsupportedOSPlatform ("macos")]
-		#if __MACCATALYST__ && !__IOS__
-		[Advice ("Starting with maccatalyst14.0 Use 'ARSession.Raycast' instead.")]
-		#endif
-		[UnsupportedOSPlatform ("maccatalyst14.0")]
+		[ObsoletedOSPlatform ("maccatalyst14.0")]
 		[SupportedOSPlatform ("ios11.0")]
 		[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
 		public virtual ARHitTestResult[] HitTest (CGPoint point, ARHitTestResultType types)
```

The generated attribute should be something like:

```cs
[ObsoletedOSPlatform ("ios14.0", "Use 'ARSession.Raycast' instead.")]
```

File: src/generator-attributes.cs [hunk 2]
```diff
@@ -1012,8 +1019,7 @@ public override string ToString ()
 			GenerateSupported (builder);
 			break;
 		case AvailabilityKind.Deprecated:
-			GenerateAdvice (builder);
```
Comment (rolfbjarne) [inline src/generator-attributes.cs:1015]: It looks like you can remove the `GenerateAdvice` method now?

File: src/generator-attributes.cs [hunk 3]
```diff
@@ -1012,8 +1019,7 @@ public override string ToString ()
...
-			GenerateUnsupported (builder);
+			GenerateDeprecated (builder);
 			break;
 		case AvailabilityKind.Obsoleted:
 			GenerateObsolete (builder);
```
Comment (rolfbjarne) [inline src/generator-attributes.cs:1025]: I'm sorry about this, but I just remembered that the `UnsupportedOSPlatform` attribute now has a `Message` property we can use to give more information, which means we don't have to use the `Obsolete` attribute for this purpose, and the call to `GenerateObsolete` can be removed as well.

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: src/generator-attributes.cs (hunk 1 above)
Comment (chamons) [reply src/generator-attributes.cs:977]: I believe I've fixed this now.

File: src/generator-attributes.cs (hunk 2 above)
Comment (chamons) [reply src/generator-attributes.cs:1015]: Done.

File: src/generator-attributes.cs (hunk 3 above)
Comment (chamons) [reply src/generator-attributes.cs:1025]: Done I believe.

**Review by dalexsoto — APPROVED** [review]

**Review by rolfbjarne — CHANGES_REQUESTED** [review]
The changes look correct, but it seems a number of tests have to be updated as well.

~Some of the monotouch-test tests are unrelated - you can fix those by merging `net7.0` into your branch.~ (the fix isn't in `net7.0` yet). There's at least one that is related though ("DelegateAndDataSourceAllowsNull: 2 failing types").

**Comment by chamons** [comment]
I'll fix `DelegateAndDataSourceAllowsNull` up and wait for you to tell me net7.0 is fixed.

**Comment by chamons** [comment]
```
[13:55:30.4847390] 	[FAIL] DelegateAndDataSourceAllowsNull : 2 failing types
[13:55:30.4848160] 2 failing types:
[13:55:30.4848270] GameKit.GKAchievementViewController: Could not initialize an instance of the type 'GameKit.GKAchievementViewController': the native 'init' method returned nil.
[13:55:30.4848380] It is possible to ignore this condition by setting ObjCRuntime.Class.ThrowOnInitFailure to false.
[13:55:30.4848470] GameKit.GKLeaderboardViewController: Could not initialize an instance of the type 'GameKit.GKLeaderboardViewController': the native 'init' method returned nil.
[13:55:30.4848550] It is possible to ignore this condition by setting ObjCRuntime.Class.ThrowOnInitFailure to false.
[13:55:30.4848630]
```

**Comment by rolfbjarne** [comment]
@chamons the unrelated failing monotouch-test tests should be fixed now.

The other test failures (cecil, introspection, xtro) all look related to the changes in this PR.

**Comment by stephen-hawley** [comment]
👀

**Review by rolfbjarne — APPROVED** [review]

**Pull request merged**